dali/test/python/test_operator_multipaste.py

```diff
@@ -0,0 +1,196 @@
+import os
+import cv2
+#from test_utils import get_dali_extra_path
+
+DEBUG_LVL = 0
+SHOW_IMAGES = False
+
+def get_dali_extra_path():
+    try:
+        dali_extra_path = os.environ['DALI_EXTRA_PATH']
+    except KeyError:
+        print("WARNING: DALI_EXTRA_PATH not initialized.", file=sys.stderr)
+        dali_extra_path = "."
+    return dali_extra_path
+
+data_root = get_dali_extra_path()
+img_dir = os.path.join(data_root, 'db', 'single', 'jpeg')
+
+
+def intersects(anchors1, shapes1, anchors2, shapes2):
+    for i in range(len(anchors1)):
+        if anchors1[i] + shapes1[i] <= anchors2[i] or anchors2[i] + shapes2[i] <= anchors1[i]:
+            return False
+    return True
+
+
+def prepare_cuts(
+        iters=4,
+        batch_size=16,
+        input_size=None,
+        output_size=None,
+        even_paste_count=False,
+        no_intersections=False
+):
+    in_idx_l = [np.zeros(shape=(0,), dtype=np.int32) for _ in range(batch_size)]
+    in_anchors_l = [np.zeros(shape=(0, 2), dtype=np.int32) for _ in range(batch_size)]
+    shapes_l = [np.zeros(shape=(0, 2), dtype=np.int32) for _ in range(batch_size)]
+    out_anchors_l = [np.zeros(shape=(0, 2), dtype=np.int32) for _ in range(batch_size)]
+    assert len(input_size) == len(output_size)
+    dim = len(input_size)
+    for i in range(batch_size):
+        for j in range(iters):
+            while True:
+                in_idx = np.int32(np.random.randint(batch_size))
+                out_idx = np.int32(i if even_paste_count else np.random.randint(batch_size))
+                shape = [np.int32(
+                    np.random.randint(
+                        min(input_size[i], output_size[i]) // (iters if no_intersections else 1)
+                    )
+                ) for i in range(dim)]
+                in_anchor = [np.int32(np.random.randint(input_size[i] - shape[i])) for i in range(dim)]
+                out_anchor = [np.int32(np.random.randint(output_size[i] - shape[i])) for i in range(dim)]
+                if no_intersections:
+                    is_ok = True
+                    for k in range(len(in_idx_l[out_idx])):
+                        if intersects(out_anchors_l[out_idx][k], shapes_l[out_idx][k], out_anchor, shape):
+                            is_ok = False
+                            break
+                    if not is_ok:
+                        continue
+                    break
+                break
+            if DEBUG_LVL > 0:
+                print(in_idx, out_idx, in_anchor, shape, out_anchor)
+            in_idx_l[out_idx] = np.append(in_idx_l[out_idx], [in_idx], axis=0)
+            in_anchors_l[out_idx] = np.append(in_anchors_l[out_idx], [in_anchor], axis=0)
+            shapes_l[out_idx] = np.append(shapes_l[out_idx], [shape], axis=0)
+            out_anchors_l[out_idx] = np.append(out_anchors_l[out_idx], [out_anchor], axis=0)
+    return in_idx_l, in_anchors_l, shapes_l, out_anchors_l
+
+
+def numpyize(source_list):
+    print(source_list)
```

Comment:
It should be dependent on DEBUG_LVL - and require a relatively high one.

Reply:
Changed